app/services/AddressSelectorService.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import models.addresslookup.{Address, AddressRecord, RecordSet}
+
+import javax.inject.Inject
+import scala.collection.SeqLike
+import scala.util.Try
+
+class AddressSelectorService @Inject() () {
+
+  def orderSet(unorderedSeq: Seq[AddressRecord]): Seq[AddressRecord] =
+    unorderedSeq.sortWith { (a, b) =>
+      def sort(zipped: Seq[(Option[Int], Option[Int])]): Boolean = zipped match {
+        case (Some(nA), Some(nB)) :: tail =>
+          if (nA == nB) sort(tail) else nA < nB
```

Comment:
not sure exactly what's going on here - is this a binary sort? Does nA refer to house number A, nB house number b?

Reply:
As simply as I can put it, It will go through the sequence of addresses sorting one tuple at a time and checking which house number (nA or nB) is bigger and whichever is bigger will be placed first. @pascal22p have I explained this well, can you add anything?

Reply:
what made you go for this method over other sorting algos? And doesn't it sort for smaller to be placed first `mkString(a.address) < mkString(b.address)` ?

Reply:
oh just saw on the ticket it links to another place this is done in that same way 👍🏽 so guess it wasn't your decision as to how to go about the sort!

Reply:
Yeah it is used in address-lookup-frontend so just used theirs and yeah it puts the smaller first sorry.

Reply:
cheers mate :)